mrp_production_raw_material_procurement_group/models/mrp_production.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 OpenSynergy Indonesia
+# Copyright 2016 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    raw_material_procurement_group_id = fields.Many2one(
+        string="Raw Material Procurement Group",
+        comodel_name="procurement.group",
+        readonly=True,
+        copy=False,
+        states={
+            "draft": [("readonly", False)],
+            },
+    )
+    auto_create_procurement_group = fields.Boolean(
+        string="Auto Create Procurement Group",
+        readonly=True,
+        states={
+            "draft": [("readonly", False)],
+            },
+        )
+
+    @api.multi
+    def action_confirm(self):
+        for mo in self:
+            mo._create_procurement_group()
+        return super(MrpProduction, self).action_confirm()
+
+    @api.multi
+    def _create_procurement_group(self):
+        self.ensure_one()
+        obj_group = self.env[
+            "procurement.group"]
+        if self.auto_create_procurement_group and \
+                not self.raw_material_procurement_group_id:
+            self.raw_material_procurement_group_id = obj_group.create({
```

Comment:
Wouldn't be better to call self._prepare_procurement_group before creating? That way we have better extensibility.